### DISCUSSION TITLE
Feature Request: LDAP Authentication for selfhosted instance

### DISCUSSION THREAD

**Discussion by fungalcofe** [plausible/analytics#635] [Anything Else]
Hi,

I would like to self-host a Plausible instance for me and a few people, but to create another account is an annoying chore.
Sure it's eased with Password managers but still annoying.

Anyway, I'd like to be able to plug an LDAP server in to authenticate already existing users (from the LDAP directory).

I have Elixir and Phoenix knowledge, so I'm willing to try to implement it.
But I'd gladly take advices and/or inputs to do it well. (and if it would merge/rebase in upstream)

Duponin

**Comment by ukutaht** [comment]
I think we can support different auth methods on the `Plug` layer with different plugins essentially. Happy to merge this upstream if you decide to build it!

**Comment by joe-eklund** [comment]
LDAP support would be great. Along those lines, support for OAuth and OIDC would be even better.

**Comment by vladimirdulov** [comment]
We are packaging Plausible app for Cloudron.io and we will be happy to see LDAP / OpenID support.

**Comment by luggesexe** [comment]
Hey, is there any update on this? I am thinking about making the switch from Matomo to Plausible but LDAP Login is needed.
